common4j/src/main/com/microsoft/identity/common/java/providers/nativeauth/responses/signup/SignUpContinueApiResponse.kt

```diff
@@ -0,0 +1,179 @@
+    @Expose @SerializedName("error") val error: String?,
+    @Expose @SerializedName("error_codes") val errorCodes: List<Int>?,
+    @Expose @SerializedName("error_description") val errorDescription: String?,
+    @SerializedName("signup_token") val signupToken: String?,
+    @Expose @SerializedName("unverified_attributes") val unverifiedAttributes: List<Map<String, String>>?,
+    @Expose @SerializedName("invalid_attributes") val invalidAttributes: List<Map<String, String>>?,
+    @Expose @SerializedName("required_attributes") val requiredAttributes: List<UserAttributeApiResult>?,
+    @SerializedName("details") val details: List<Map<String, String>>?
+) : IApiResponse(statusCode) {
+
+    companion object {
+        private val TAG = SignUpContinueApiResponse::class.java.simpleName
+    }
+
+    fun toResult(): SignUpContinueApiResult {
+        LogSession.logMethodCall(TAG, "${TAG}.toResult")
+
+        return when (statusCode) {
+
+            // Handle 400 errors
+            HttpURLConnection.HTTP_BAD_REQUEST -> {
+                return when {
+                    error.isPasswordTooWeak() || error.isPasswordTooLong() || error.isPasswordTooShort() || error.isPasswordBanned() ||
+                            error.isPasswordRecentlyUsed() -> {
+                        SignUpContinueApiResult.InvalidPassword(
+                            error = error.orEmpty(),
+                            errorDescription = errorDescription.orEmpty()
+                        )
+                    }
+                    error.isUserAlreadyExists() -> {
+                        SignUpContinueApiResult.UsernameAlreadyExists(
+                            error = error.orEmpty(),
+                            errorDescription = errorDescription.orEmpty()
+                        )
+                    }
+                    error.isAttributeValidationFailed() -> {
+                        SignUpContinueApiResult.InvalidAttributes(
+                            error = error.orEmpty(),
+                            errorDescription = errorDescription.orEmpty(),
+                            invalidAttributes = invalidAttributes?.toAttributeList()
+                                ?: return SignUpContinueApiResult.UnknownError(
+                                    error = "invalid_state",
+                                    errorDescription = "SignUp /continue did not return a invalid_attributes with validation_failed error",
+                                    details = details
+                                )
+                        )
+                    }
+                    error.isExpiredToken() -> {
+                        SignUpContinueApiResult.ExpiredToken(
+                            error = error.orEmpty(),
+                            errorDescription = errorDescription.orEmpty()
+                        )
+                    }
+                    (error.isInvalidRequest() and errorCodes?.get(0).isOtpCodeIncorrect()) ->{
```

Comment:
Some comments in the code explaining what's happening would certainly help

Reply:
I have added a comment